pkg/datalayer/datalayer.go

```diff
@@ -21,10 +22,50 @@ func VolumeExists(basePath string, volumeName string) bool {
 	return err == nil
 }
 
+func ActiveVolume(basePath string) (string, error) {
+	currentVolumeJsonPath := filepath.FromSlash(basePath + "/" + "current_volume.json")
+	file, err := os.Open(currentVolumeJsonPath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	var store map[string]interface{}
+	err = decoder.Decode(&store)
+	if err != nil {
+		return "", err
+	}
+	return store["current_volume"].(string), nil
+}
+
+func setActiveVolume(basePath, volumeName string) error {
+	currentVolumeJsonPath := filepath.FromSlash(basePath + "/" + "current_volume.json")
+	currentVolumeContent := map[string]string{
+		"current_volume": volumeName,
+	}
+	// Create or update this file
+	file, err := os.Create(currentVolumeJsonPath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	encoder := json.NewEncoder(file)
+	encoder.Encode(currentVolumeContent)
+	return nil
+}
+
 func CreateVolume(basePath string, volumeName string) error {
 	volumePath := filepath.FromSlash(basePath + "/" + volumeName)
 	// TODO Factor this into a data layer object.
-	return os.MkdirAll(volumePath, 0777) // XXX SEC
+	err := os.MkdirAll(volumePath, 0777) // XXX SEC
+	if err != nil {
+		return err
+	}
```

Comment:
the following lines can be simplified to `return setActiveVolume(basePath, volumeName)`

(currently it smells a bit like `if x return true else return false` which can be simplified to `return x` :smile:)